jupyter_server_fileid/manager.py

```diff
@@ -120,30 +124,35 @@ def __init__(self, *args, **kwargs):
         self.con.execute("PRAGMA journal_mode = WAL")
         self.con.execute(
             "CREATE TABLE IF NOT EXISTS Files("
-            "id INTEGER PRIMARY KEY AUTOINCREMENT, "
+            "id TEXT PRIMARY KEY NOT NULL, "
```

Comment:
Besides storing `root_dir` (mentioned in the review on #24), an indication of whether or not the row corresponds to a directory should also be persisted.

Reply:
Again, this requires assumptions of the filesystem. The key is that `ArbitraryFileIdManager` will work with literally any contents manager that emits events.